redash/models.py

```diff
@@ -757,6 +776,190 @@ def __unicode__(self):
+    TYPE_MODIFY = 'modify'
+
+    class Meta:
+        db_table = 'changes'
+
+    def to_dict(self, full=True):
+        d = {
+            'id': self.id,
+            'object_id': self.object_id,
+            'object_type': self.object_type,
+            'change_type': self.change_type,
+            'object_version': self.object_version,
+            'change': self.change,
+            'created_at': self.created_at
+        }
+
+        if full:
+            d['user'] = self.user.to_dict()
+        else:
+            d['user_id'] = self.user_id
+
+        return d
+
+    @staticmethod
+    def save_change(user, new_object, object_id=None, object_type=None,
+            old_object=None, change_type=None, object_version=None):
+        # determine change type
+        if not change_type:
+            if old_object is None:
+                change_type = Change.TYPE_CREATE
+            else:
+                change_type = Change.TYPE_MODIFY
+        # determine object type
+        if not object_type and new_object is not None:
+            object_type = new_object.__class__.__name__
+        # determine object id
+        if not object_id and new_object is not None:
+            object_id = new_object.id
+        if not object_version and new_object is not None:
+            object_version = new_object.version
+        # make sure objects are dicts
+        if not isinstance(new_object, dict):
+            new_object = new_object.to_dict()
+        if old_object and not isinstance(old_object, dict):
+            old_object = old_object.to_dict()
+        # remove dates from objects, convert obj references to IDs
+        for obj in [old_object, new_object]:
+            if obj:
+                for field in list(obj.keys()):
+                    val = obj[field]
+                    if isinstance(val, datetime.datetime):
+                        del obj[field]
+                    elif isinstance(val, BaseModel):
+                        obj[field] = obj[field].id
+                    elif isinstance(val, dict) and 'id' in val:
+                        obj[field] = obj[field]['id']
+        # create and save Change
+        change = Change()
+        change.object_id = object_id
+        change.object_type = object_type
+        change.change_type = change_type
+        change.object_version = object_version
+        change.user = user
+        change.change = {
+            "before": old_object,
+            "after": new_object
+        }
+        change.save()
+        return change
+
+    @staticmethod
+    def get_latest(object_id, object_type, change_type=None):
+        try:
+            query = Change.select(Change)\
+                .where(Change.object_id == object_id)\
+                .where(Change.object_type == object_type)
+            if change_type:
+                query = query.where(Change.change_type == change_type)
+            query = query.order_by(Change.created_at.desc())
+            for change in query:
```

Comment:
It will query the database for all the changes, create objects for them and then use only the first...

Just do: `return query.limit(1).first()` (`first()` by itself doesn't limit the result set).